components.json

```diff
@@ -651,6 +651,14 @@
 			"require": "markup",
 			"owner": "Golmote"
 		},
+		"tt2": {
+			"title": "Template Toolkit",
+			"alias": ["tt"],
+			"aliasTitles": {
+				"tt": "Template Toolkit"
+			},
+			"owner": "gflohr"
+		},
```

Comment:
I would suggest using the name `tt` as the main name, since it seems to be the more generic (if I understand correctly, `tt2` would be the second version?). If you do so, you'll need to rename the component file too.

Reply:
Also, starting with Prism 1.14, we made it more simple to craft templating languages using the placeholders `___...___`.
To benefit from this, your component should `"require": "markup-templating"` here. That way, you'll have access to utilities functions to automate the placeholders creation/replacement process.

Have a look at the [ERB](https://github.com/PrismJS/prism/blob/gh-pages/components/prism-erb.js), [Handlebars](https://github.com/PrismJS/prism/blob/gh-pages/components/prism-handlebars.js), [PHP](https://github.com/PrismJS/prism/blob/gh-pages/components/prism-php.js) or [Smarty](https://github.com/PrismJS/prism/blob/gh-pages/components/prism-smarty.js) components for examples. You can also take a look at the functions in the [Markup-templating](https://github.com/PrismJS/prism/blob/gh-pages/components/prism-markup-templating.js) component.

The way to create/replace the placeholders changed a bit, but from your component perspective, it should only make it simpler, really.

Reply:
Sounds good to me. No alias at all, use `tt2` as the identifier. Maybe just simplify the title to "Template Toolkit 2" :wink:

Reply:
Agreed, I will just use "Template Toolkit 2".

Reply:
I am rewriting the component from scratch now but I am not sure that I am completely on the right track.

A typical example for a TT2 template is https://github.com/gflohr/qgoda-site/blob/master/_views/components/listing.html.

My current component is here: https://github.com/gflohr/prism/blob/gh-pages/components/prism-tt2.js

Two features are causing me a headache:

```
[%# This is a multi-line comment
    because the "#" is immediately following
    the opening tag.
%]
[% # And this is a normal comment.
    x = 42 # So is this.
%]
```

I tried to implement that morelike this:

```javascript
Prism.languages.insertBefore('tt2', 'delimiter', {
	comment: /\[%#[^]*?\%\]/
});
```

But at no avail.  Only the first line is recognized as comment.

And as an alternative to `[% ... %]` you can use [outline tags](http://www.template-toolkit.org/docs/manual/Syntax.html#section_Outline_Tags):

```
%% IF foo = "bar"
<br />
%% END
```

The "closing delimiter" is a newline here.  How can I specify that?

Both features are a little bit esoteric, and could also be left as a known limitation if there is no solution.